### PR TITLE
Use convertToZeroTerminatedString on both wasm and js targets

### PR DESCRIPTION
**Pull request by Schahen**
This is supposed to be fix for https://github.com/JetBrains/skiko/issues/967

The idea is pass strings as (utf8) byte arrays directly through the border
just like we do it on native

### CODE REVIEW

File: skiko/src/nativeJsMain/kotlin/org/jetbrains/skia/impl/Native.nativejs.kt [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+/**
+ * Converts String to zero-terminated utf-8 byte array.
+ */
+internal fun convertToZeroTerminatedString(string: String): ByteArray {
+    //  C++ needs char* with zero byte at the end. So we need to copy array with an extra zero byte.
+
+    val utf8 = string.encodeToByteArray() // encodeToByteArray encodes to utf8
+    // TODO Remove array copy, use `skString(data, length)` instead of `skString(data)`
+    return utf8.copyOf(utf8.size + 1)
```
Comment (eymar) [inline skiko/src/nativeJsMain/kotlin/org/jetbrains/skia/impl/Native.nativejs.kt:11]: Is it guaranteed that the 1 extra byte would be `\0`? Or maybe we better set its value explicitly?

Comment (Schahen) [reply skiko/src/nativeJsMain/kotlin/org/jetbrains/skia/impl/Native.nativejs.kt:11]: it's guaranteed by  spec for  ByteArray.copyOf from Kotlin stdlib

File: skiko/src/commonTest/kotlin/org/jetbrains/skia/ManagedStringTest.kt [hunk 2]
```diff
@@ -23,6 +23,12 @@ class ManagedStringTest {
         assertEquals("你好", s3)
     }
 
+    @Test
+    fun emptyStringTest() {
```
Comment (eymar) [inline skiko/src/commonTest/kotlin/org/jetbrains/skia/ManagedStringTest.kt:27]: What do you think about adding a few more simple tests with intiially empty strings:

`s.insert(0, "")`, `s.append("")`.

Also I'm wondering what is going to happen if we call insert with an index out of bounds. Also `remove` with an index out of bounds.

**Review by eymar — APPROVED** [review]

File: skiko/src/nativeJsMain/kotlin/org/jetbrains/skia/impl/Native.nativejs.kt [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package org.jetbrains.skia.impl
+
+/**
+ * Converts String to zero-terminated utf-8 byte array.
+ */
+internal fun convertToZeroTerminatedString(string: String): ByteArray {
```
Comment (MatkovIvan) [inline skiko/src/nativeJsMain/kotlin/org/jetbrains/skia/impl/Native.nativejs.kt:6]: The name is confusing - I thought that `toInterop` is recursive. It converts to `ByteArray`, not string

Also better to include "utf8" in some form to the name
